FinalProject/gs-rest-service-master/complete/src/main/java/services/Imp/CartServiceImp.java

```diff
@@ -0,0 +1,49 @@
+package services.Imp;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import services.CartService;
+import bo.Cart;
+import bo.CartItem;
+import bo.User;
+import dao.CartDao;
+import dao.imp.CartDaoImp;
+
+public class CartServiceImp implements CartService {
+
+//	@Autowired
+	CartDao cartDao = new CartDaoImp();
+
+	public Cart addNewCart(User user) throws SQLException {
+		int cartId = cartDao.getLastId();
+		cartId++;
+		Cart cart = new Cart(cartId, user);
+		cartDao.addCart(cart);
+		return cart;

```

Comment:
Why you return the cart that you create in memory? What happend if the cart never persist for any reason?